pytest-xdist race conditions

```
=================================== FAILURES ===================================
________________________________ test_list_dids ________________________________
[gw1] linux -- Python 3.12.5 /usr/local/bin/python3.12

governance_client = <shared.util.rich_async_client.RichAsyncClient object at 0x7f84aab46900>
mock_governance_auth = <DummyAcaPyAuthVerified id=140207776361808>

    @pytest.mark.anyio
    async def test_list_dids(
        governance_client: RichAsyncClient, mock_governance_auth: AcaPyAuthVerified
    ):
        response = await governance_client.get(WALLET_BASE_PATH)
    
        assert response.status_code == 200
        response = response.json()
    
        res_method: List[DID] = await list_dids(auth=mock_governance_auth)
        res_method_dict = list(map(lambda x: x.to_dict(), res_method))
>       assert res_method_dict == response
E       AssertionError: assert [{'did': 'Bo9...v', ...}, ...] == [{'did': 'Bo9...v', ...}, ...]
E         
E         At index 79 diff: {'did': 'JRX8acV5BCxBPhSUiA5Z5s', 'key_type': 'ed25519', 'metadata': {}, 'method': 'sov', 'posture': 'wallet_only', 'verkey': 'AVmWmtwSpBYMruPBf8NMkTgQADLYcp4Mbawo5n71osMu'} != {'did': 'KSiagjE5BAruDJpeSU34ce', 'key_type': 'ed25519', 'metadata': {}, 'method': 'sov', 'posture': 'wallet_only', 'verkey': 'B42ruDEiLpyYFb8Wk5rGyzMXoTiWtLWfgWN3FKS3Ww8e'}
E         Left contains one more item: {'did': 'owLxNjGRuF36TMeDtgs64', 'key_type': 'ed25519', 'metadata': {}, 'method': 'sov', ...}
E         Use -v to get more diff

app/tests/e2e/test_wallet_dids.py:47: AssertionError
---------------------------- Captured stdout setup -----------------------------
2024-08-12 19:12:05.028 | INFO     | app.tests.e2e.conftest:governance_public_did:42 | Configuring public did for governance | 
2024-08-12 19:12:05.028 | INFO     | app.services.acapy_wallet:get_public_did:114 | Fetching public DID | 
2024-08-12 19:12:05.079 | INFO     | app.services.acapy_wallet:get_public_did:123 | Successfully fetched public DID. | 
----------------------------- Captured stdout call -----------------------------
2024-08-12 19:12:05.204 | INFO     | app.routes.wallet.dids:list_dids:47 | GET request received: Retrieve list of DIDs | 
2024-08-12 19:12:05.360 | INFO     | app.routes.wallet.dids:list_dids:59 | Successfully fetched list of DIDs. | 
=============================== warnings summary ===============================
app/tests/e2e/test_proof_request_models.py::test_proof_model_failures[clean-clean-clean-clean-clean-clean-Proof-None-v1]
app/tests/e2e/test_proof_request_models.py::test_proof_model_failures[clean-clean-clean-clean-clean-clean-None-1.0-v1]
app/tests/e2e/test_proof_request_models.py::test_proof_model_failures[clean-clean-clean-clean-clean-clean-None-None-v1]
  /usr/local/lib/python3.12/site-packages/aries_cloudcontroller/api/present_proof_v10_api.py:1168: DeprecationWarning: POST /present-proof/send-request is deprecated.
    warnings.warn(

app/tests/e2e/test_proof_request_models.py::test_proof_model_failures[clean-clean-clean-clean-clean-clean-Proof-None-v1]
app/tests/e2e/test_proof_request_models.py::test_proof_model_failures[clean-clean-clean-clean-clean-clean-None-1.0-v1]
app/tests/e2e/test_proof_request_models.py::test_proof_model_failures[clean-clean-clean-clean-clean-clean-None-None-v1]
  /usr/local/lib/python3.12/site-packages/aries_cloudcontroller/api/present_proof_v10_api.py:194: DeprecationWarning: DELETE /present-proof/records/{pres_ex_id} is deprecated.
    warnings.warn(

-- Docs: https://docs.pytest.org/en/stable/how-to/capture-warnings.html
=========================== short test summary info ============================
FAILED app/tests/e2e/test_wallet_dids.py::test_list_dids - AssertionError: as...
====== 1 failed, 1065 passed, 4 skipped, 6 warnings in 181.32s (0:03:01) =======
```

--- a/app/tests/e2e/issuer/test_save_exchange_record.py
+++ b/app/tests/e2e/issuer/test_save_exchange_record.py
@@ -73,9 +73,7 @@ async def test_issue_credential_with_save_exchange_record(
             },
         )
 
-        time.sleep(
-            1
-        )  # short sleep before fetching cred ex records; allow them to update
+        await asyncio.sleep(1)  # short sleep before fetching; allow records to update
 
         # faber requesting auto_remove only removes their cred ex records
         # get exchange record from alice side -- should not exist after complete

--- a/app/tests/e2e/test_wallet_dids.py
+++ b/app/tests/e2e/test_wallet_dids.py
@@ -37,14 +37,23 @@ async def create_did_mock(governance_client: RichAsyncClient):
 async def test_list_dids(
     governance_client: RichAsyncClient, mock_governance_auth: AcaPyAuthVerified
 ):
+    # Capture the existing DIDs before the request
+    initial_dids = await list_dids(auth=mock_governance_auth)
+    initial_dids_set = set(map(lambda x: x.did, initial_dids))
+
+    # Make the GET request
     response = await governance_client.get(WALLET_BASE_PATH)
-
     assert response.status_code == 200
-    response = response.json()
+    response_data = response.json()
 
-    res_method: List[DID] = await list_dids(auth=mock_governance_auth)
-    res_method_dict = list(map(lambda x: x.to_dict(), res_method))
-    assert res_method_dict == response
+    # Filter the response to include only the initial DIDs
+    filtered_response_data = [
+        did_dict for did_dict in response_data if did_dict["did"] in initial_dids_set
+    ]
+
+    # Compare the filtered response with the initial DIDs
+    initial_dids_dict = list(map(lambda x: x.to_dict(), initial_dids))
+    assert filtered_response_data == initial_dids_dict
 
 
 @pytest.mark.anyio

--- a/app/tests/e2e/verifier/test_verifier.py
+++ b/app/tests/e2e/verifier/test_verifier.py
@@ -646,6 +646,7 @@ async def test_saving_of_presentation_exchange_records(
     assert isinstance(pres_exchange_result, PresentationExchange)
     assert response.status_code == 200
 
+    await asyncio.sleep(1)  # short sleep before fetching records; allow them to update
     # get exchange records from alice side
     if alice_save_exchange_record:
         # Save record is True, should be 1 record

--- a/app/tests/e2e/verifier/test_verifier.py
+++ b/app/tests/e2e/verifier/test_verifier.py
@@ -630,6 +630,11 @@ async def test_saving_of_presentation_exchange_records(
         VERIFIER_BASE_PATH + "/accept-request",
         json=proof_accept.model_dump(),
     )
+    result = response.json()
+
+    pres_exchange_result = PresentationExchange(**result)
+    assert isinstance(pres_exchange_result, PresentationExchange)
+    assert response.status_code == 200
 
     await assert_both_webhooks_received(
         alice_member_client,
@@ -639,12 +644,6 @@ async def test_saving_of_presentation_exchange_records(
         alice_proof_id,
         acme_proof_id,
     )
-
-    result = response.json()
-
-    pres_exchange_result = PresentationExchange(**result)
-    assert isinstance(pres_exchange_result, PresentationExchange)
-    assert response.status_code == 200
 
     await asyncio.sleep(1)  # short sleep before fetching records; allow them to update
     # get exchange records from alice side

